app/services/user_seeder.rb

```diff
@@ -0,0 +1,84 @@
+class UserSeeder
+  def self.run(**opts)
+    new(**opts).run
+  end
+
+  def initialize(**options)
+    @csv_file = options[:csv_file]
+    @csv = CSV.parse(File.read(options[:csv_file]), headers: true)
+    @email_domain = options[:email_domain]
+  end
+
+  def run
+    puts "Running with CSV #{csv_file} and domain #{email_domain}"
+
+    @csv.each_with_index do |row, i|
+      email = "user#{i}@#{email_domain}"
+      row['email_address'] = email
+      row['password'] = PASSWORD
+      ee = EncryptedAttribute.new_from_decrypted(email)
+
+      User.find_or_create_by!(email_fingerprint: ee.fingerprint) do |user|
+        codes = setup_user(user: user, ee: ee)
+        row['codes'] = codes.join('|')
+
+        profile = Profile.new(user: user)
+        pii = Pii::Attributes.new_from_hash(
```

Comment:
Is there a cleaner way to do this? Maybe convert to a hash, slice with the appropriate keys, and then transform keys.